image_finder.py

```diff
@@ -81,65 +76,127 @@ def find_circle_in_image(image_frame: video_stream.TimedFrame, app_context: app_
         # print(error)
     return None, None
 
+
+
+def blobFilter(image_frame:numpy.ndarray, app_context: app_contexts.AppContext):
+    params = cv2.SimpleBlobDetector_Params() 
+        
+
+    # Set Area filtering parameters 
+    params.filterByArea = True
+    params.minArea = app_context.circle_context.min_area
+    params.maxArea = 10000
+
+    # Set Circularity filtering parameters 
+    params.filterByCircularity = True 
+    params.minCircularity = app_context.circle_context.min_circularity
+    params.maxCircularity = 1
+
+    # Set Convexity filtering parameters 
+    params.filterByConvexity = True
+    params.minConvexity = app_context.circle_context.min_convexity
+
+    # Set inertia filtering parameters 
+    params.filterByInertia = True
+    params.minInertiaRatio = app_context.circle_context.min_inertia
+    params.filterByColor = False
+    # Create a detector with the parameters 
+    detector = cv2.SimpleBlobDetector_create(params) 
+        
+    # Detect blobs 
+    keypoints = detector.detect(image_frame) 
+    return keypoints
+
 def find_image(image_frame: video_stream.TimedFrame, app_context: app_contexts.AppContext) -> typing.Tuple[typing.Optional[numpy.ndarray], typing.Optional[video_stream.TimedFrame]]:
     try:
+
+       
         start_time = time.time()
         if image_frame.frame is None:
+            logger.debug("Skipping empty image")
             return None, image_frame
+        
+        if(app_context.color_context.orientation == app_contexts.Orientation.UPSIDE_DOWN):
```

Comment:
nit: if formatting is inconsistent with standard python. If you want to, use a formatter like [black](https://github.com/psf/black)